source/server-discovery-and-monitoring/tests/monitoring/standalone_debounce_description_updates.yml

```diff
@@ -0,0 +1,73 @@
+description: "Monitoring a standalone connection - multiple equal description updates are debounced"
+uri: "mongodb://a:27017"
+phases:
+  -
+    responses:
+      -
+        - "a:27017"
+        - { ok: 1, ismaster: true, minWireVersion: 0, maxWireVersion: 4 }
+      -
+        - "a:27017"
+        - { ok: 1, ismaster: true, minWireVersion: 0, maxWireVersion: 4 }
+
+    outcome:
+      events:
+        -
+          topology_opening_event:
+            topologyId: "42"
+        -
+          topology_description_changed_event:
+            topologyId: "42"
+            previousDescription:
+              topologyType: "Unknown"
+              servers: []
+            newDescription:
+              topologyType: "Single"
+              servers:
+                -
+                  address: "a:27017"
+                  arbiters: []
+                  hosts: []
+                  passives: []
+                  type: "Unknown"
+        -
+          server_opening_event:
+            topologyId: "42"
+            address: "a:27017"
+        -
+          server_description_changed_event:
+            topologyId: "42"
+            address: "a:27017"
+            previousDescription:
+              address: "a:27017"
+              arbiters: []
+              hosts: []
+              passives: []
+              type: "Unknown"
+            newDescription:
+              address: "a:27017"
+              arbiters: []
+              hosts: []
+              passives: []
+              type: "Standalone"
+        -
+          topology_description_changed_event:
```

Comment:
I haven't worked with SDAM spec tests before, but does the following quote from [Test Phases](https://github.com/mongodb/specifications/tree/master/source/server-discovery-and-monitoring/tests#test-phases) ensure that the observed events cannot exceed those included in `outcome.events`?

> For monitoring tests, once all responses are processed, assert that the events collected so far by the SDAM event listener are _equivalent_ to the events specified in the phase.

I wanted to ask, as our APM assertions in other operation-based tests would typically allow (and ignore) extra events to be emitted beyond what's in the spec tests.

If so, is the purpose of this test to assert that a second `server_description_changed_event` and/or `topology_description_changed_event` event is not emitted with identical data?

Reply:
The specification calls for exact match between emitted events and the expected events.

In Ruby we sometimes explicitly exclude certain events from the checks (such as authenticate and sasl* commands), but in general we do perform equality comparison and not just a subset.

Reply:
Yes, the purpose of this test is to show that only one set of events is emitted given the same input